### ISSUE TITLE
Fails to install the desktop file: Permission denied

### ISSUE THREAD

**Issue by yurivict** [CsoundQt/CsoundQt#289]
On FreeBSD I get this while trying to update the port:

```
cd /wrkdirs/usr/ports/audio/csoundqt/work/CsoundQt-0.9.6-14-g2aa6c5f9/bin/; ln -sf CsoundQt-d-cs6 csoundqt
rm -f /usr/local/share/applications/CsoundQt.desktop && desktop-file-install --mode=644 --dir=/usr/local/share/applications /wrkdirs/usr/ports/audio/csoundqt/work/CsoundQt-0.9.6-14-g2aa6c5f9/csoundqt.desktop
Error on file "/wrkdirs/usr/ports/audio/csoundqt/work/CsoundQt-0.9.6-14-g2aa6c5f9/csoundqt.desktop": Failed to create file ?/usr/local/share/applications/csoundqt.desktop.FIV7D0?: Permission denied
```

0.9.6-14-g2aa6c5f9

**Comment by tarmoj** [comment]
Hi,

seems that it comes from the line:

` desktop-file-install --mode=644 --dir=/usr/local/share/applications /wrkdirs/usr/ports/audio/csoundqt/work/CsoundQt-0.9.6-14-g2aa6c5f9/csoundqt.desktop`

That is generated by qmake from  qcs.pro  where the according line is:
`desktop.commands += desktop-file-install --mode=644 --dir=$$SHARE_DIR/applications $$PWD/csoundqt.desktop`

I don't know about the building and installing process, can you set the SHARE_DIR as needed in your system like: 

`qmake qcs.pro INSTALL_DIR=~  SHARE_DIR=~/.local/share`

or whatever is more consistent for you.

Hope it helps.

tarmo

**Comment by yurivict** [comment]
These are already set: ```INSTALL_DIR=${PREFIX} SHARE_DIR=${PREFIX}/share```

**Comment by tarmoj** [comment]
And I assume ${PREFIX} is set to  /usr/local/ ?
Then it makes sense.
Normally on Linux you would do `sudo make install` then  there is no problem with the permissions.
In my generated makefile the line looks like:
`rm -f /usr/share/applications/CsoundQt.desktop && desktop-file-install --mode=644 --dir=/usr/share/applications /home/tarmo/src/qutecsound/qutecsound-code/csoundqt.desktop`

Sorry, I have never been into packaging, perhaps these build services can give an example:
https://build.opensuse.org/package/show/home:geekositalia:daw/CsoundQt
https://build.opensuse.org/package/show/home:zhonghuaren/csoundqt

tarmo

tarmo

tarmo

**Comment by yurivict** [comment]
No, no 'sudo', this is just for users as a convenience feature breaking security.

It should install everything into $(DESTDIR)$(PREFIX) without any privilege level escalation.

**Comment by yurivict** [comment]
I am not familiar with SuSe - it is a closed commercial system.

But CsoundQt is broken, it fails to install the desktop file.

**Comment by yurivict** [comment]
Installation should occur into a stage directory, which is ```$(DESTDIR)```. Installing just into /usr/local isn't sufficient and isn't possible.

**Comment by tarmoj** [comment]
To me then
INSTALL_DIR=$(DESTDIR)/$(PREFIX) SHARE_DIR=$(DESTDIR)/$(PREFIX)/share
would seem most logical.
tarmo

L, 4. jaanuar 2020 21:08 yuri@FreeBSD <notifications@github.com (mailto:notifications@github.com)> kirjutas:

**Comment by tarmoj** [comment]
Fixed in v0.9.7
